src/cmd/cli/command/estimate.go

```diff
@@ -55,29 +53,26 @@ func makeEstimateCmd() *cobra.Command {
 	return estimateCmd
 }
 
-func RunEstimate(ctx context.Context, project *compose.Project, provider cliClient.Provider, region string, mode defangv1.DeploymentMode) error {
+func RunEstimate(ctx context.Context, project *compose.Project, provider cliClient.ProviderID, region string, mode defangv1.DeploymentMode) (*defangv1.EstimateResponse, error) {
+	defangProvider := &cliClient.PlaygroundProvider{FabricClient: client}
 	term.Info("Generating deployment preview")
-	preview, err := GeneratePreview(ctx, project, provider, mode)
+	preview, err := GeneratePreview(ctx, project, defangProvider, mode)
 	if err != nil {
-		return fmt.Errorf("failed to generate preview: %w", err)
+		return nil, fmt.Errorf("failed to generate preview: %w", err)
```

Comment:
User doesn't know what "generate preview" means. This err feels too low level.

Reply:
```suggestion
		return nil, err
```